### PR TITLE
omelasticsearch: add dynamic pipeline support

### PR DESCRIPTION
**Pull request by WaeCo**
Hello,

in addition to #1862 this adds dynamic configurable pipeline support using templates.
This PR requires #1862 so I included it here.

I also did some cleanup of the template requesting logic.

Example Configuration:
```
template(name="pipelineNameTpl" type="string" string="%msg%")

action(type="omelasticsearch"
       server="localhost:9200"
       searchIndex="rsyslog-test"
       pipelineName="pipelineNameTpl"
       dynPipelineName="on")
```

### CODE REVIEW

**Comment by rgerhards** [comment]
thx. Just so that you know: we currently have issues with omelasticsearch and I'll merge only after these are solved (to reduce complexity). I may ask you to rebase to master to get the updated code before I can merge.

**Comment by rgerhards** [comment]
FYI: In addition to fixing the issues, it would also be good to rebase to current master. The freebsd CI machine has more checks active, but this requires support from new master (else it fails, as it did here). You were just a couple of hours too fast to get the necessary changes in master ;-)

**Comment by rgerhards** [comment]
would you mind updating the doc as well?

**Comment by WaeCo** [comment]
doc update: rsyslog/rsyslog-doc#384

**Comment by rgerhards** [comment]
@WaeCo could you do a rebase to current master? That would be great. Else I'll see that I merge via a dedicated merge PR.

**Comment by rgerhards** [comment]
I should mention that I just merged the big bunches of ES fixes I did and am now ready to go ahead to merge the feature-enhancement PRs - thus the question on rebase.

File: plugins/omelasticsearch/omelasticsearch.c [hunk 1]
```diff
@@ -570,6 +573,10 @@ computeMessageSize(const wrkrInstanceData_t *const pWrkrData,
 	uchar *pipelineName;
 
 	getIndexTypeAndParent(pWrkrData->pData, tpls, &searchIndex, &searchType, &parent, &bulkId, &pipelineName);
+DBGPRINTF("computeMessageSize: message %p, searchIndex %p, searchType%p\n", message, searchIndex, searchType);
```
Comment (Niels-Be) [inline plugins/omelasticsearch/omelasticsearch.c]: Are those debug messages sill required?

Comment (rgerhards) [reply plugins/omelasticsearch/omelasticsearch.c]: For the time being, yes (we are still looking at some issue).

**Pull request merged**